odl/util/utility.py

```diff
@@ -147,3 +128,89 @@ def __new__(cls, name, this_bases, d):
                 return type.__new__(cls, name, (), d)
             return meta(name, bases, d)
     return metaclass('temporary_class', None, {})
+
+
+def is_scalar_dtype(dtype):
+    """Whether a datatype is scalar or not."""
+    return np.issubsctype(dtype, np.number)
+
+
+def is_int_dtype(dtype):
```

Comment:
Kinda skeptical to these functions overall. They dont provide much and simply wrap a numpy function. How about remove? (Not sure here).